### PR TITLE
test: Lowpan module unit tests

### PR DESCRIPTION
**Pull request by turon**
Added initial unit tests for 6LoWPAN layer decompression. 
Extended unit test framework with helper functions for setting up ASCI hex test vectors.

### CODE REVIEW

File: tests/unit/Makefile.am [hunk 1]
```diff
@@ -55,6 +57,7 @@ AM_CPPFLAGS                                                    = \
 
 COMMON_LDADD                                                   =  \
     $(top_builddir)/src/core/libopenthread.a                      \
+    $(top_builddir)/src/cli/libopenthread-cli.a 		  \
```
Comment (jwhui) [inline tests/unit/Makefile.am]: Is this needed?

File: tests/unit/test_lowpan.cpp [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+extern "C" void otSignalTaskletPending(void) {}
+
+extern "C" bool otAreTaskletsPending(void) { return false; }
+
+extern "C" void otPlatSerialSendDone(void) {}
+
+extern "C" void otPlatSerialReceived(const uint8_t *aBuf, uint16_t aBufLength) 
+{
+}
```
Comment (jwhui) [inline tests/unit/test_lowpan.cpp]: Make `{` and `}` formatting consistent with above?

File: tests/unit/test_lowpan.cpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        message->Write(Ip6::Header::GetPayloadLengthOffset(),
+                       sizeof(ip6PayloadLength), &ip6PayloadLength);
+
+        resultLength = message->GetLength();
+        message->Read(0, resultLength, result);
+
+        printf("Decompressed OpenThread:\n");
+        otTestPrintHex(result, resultLength);
+
+        SuccessOrQuit(memcmp(ipVector.data(), result, resultLength),
```
Comment (jwhui) [inline tests/unit/test_lowpan.cpp]: Should probably use `VerifyOrQuit` instead.

File: tests/unit/test_lowpan.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                      "6lo: Lowpan::Decompress failed");
+
+        // ===> Test Lowpan::Compress
+        int resultLength = sMockLowpan.Compress(*message, macSource, macDest,
+                                                result);
+        printf("Compressed OpenThread:\n");
+        otTestPrintHex(result, resultLength);
+
+        SuccessOrQuit(memcmp(frame.GetPayload(), result, resultLength),
```
Comment (jwhui) [inline tests/unit/test_lowpan.cpp]: Should probably use `VerifyOrQuit` instead.

File: tests/unit/test_util.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    std::string word;
+
+    while (ss >> word)
+    {
+        uint8_t n = strtol(word.data(), NULL, 16);
+        aOutBytes.push_back(n);
+    }
+}
+
+void otTestPrintHex(uint8_t *aBuffer, int aLength)
```
Comment (jwhui) [inline tests/unit/test_util.cpp]: What about using `otDump` instead?

Comment (turon) [inline tests/unit/test_util.cpp]: This one is designed to provide corollary output to otTestHexToVector for easy visual comparison.  Also, it is designed to always output as unit test framework wants a known stable logging level.  I can look into converging to otDump if needed.  Alternatively, output of result and expected can be removed.

**Comment by jwhui** [comment]
LGTM 👍

**Pull request merged**